### DISCUSSION TITLE
How to clear a selected option field using js ?

### DISCUSSION THREAD

**Discussion by emb-karan** [codingjoe/django-select2#24] [Q&A]
I am trying to clear a selected option using js using the following code
`$('#select').select2({
  placeholder: '',
  allowClear: true
});`
and
`$("#select2-id-xyz-container").val(null).trigger("change");`
That resets the data only over the UI but when I click on submit, cleared data  may be still present and got saved

**Comment by codingjoe** [comment]
Hi @emb-karan,

This seems to be related to the Select2 JavaScript package, not `django-select2`, or is it?
Anyhow, `django-select2` handles this automatically. All you need to do is make the form field as `required=False`.
Besides, from our implementation I know that you need to add an empty option, to make a select2 input clearable.

I hope my answer helps you.

Best,
Joe
